src/util/prov_helpers.rs

```diff
@@ -1,39 +1,279 @@
+use cosmwasm_std::{from_json, CustomQuery, Uint64};
+use provwasm_std::types::{
+    cosmos::base::query::v1beta1::{PageRequest, PageResponse},
+    provenance::{
+        attribute::v1::{AttributeQuerier, AttributeType, QueryAttributesResponse},
+        name::v1::{MsgBindNameRequest, NameRecord},
+    },
+};
+use result_extensions::ResultExtensions;
+
+use crate::types::core::error::ContractError;
 
 /// Parses all group ids from the [Provenance Blockchain Attributes](https://docs.provenance.io/modules/account)
 /// provided by filtering for all values that match the given name and have an assigned int value.
 ///
 /// # Parameters
 ///
-/// * `attributes` Attributes fetched via a chain query.
+/// * `attributes` Pages of Attributes fetched via a chain query.
 /// * `name` A [Provenance Blockchain Name Module](https://docs.provenance.io/modules/name-module)
 /// name used to write the attribute.
-pub fn get_group_id_attribute_values<S: Into<String>>(
-    attributes: &Attributes,
+pub fn get_group_id_attribute_values_paginated<S: Into<String>>(
+    attributes: Vec<QueryAttributesResponse>,
     name: S,
 ) -> Vec<Uint64> {
     let name = name.into();
     attributes
-        .attributes
         .iter()
-        .filter(|attr| attr.name == name && attr.value_type == AttributeValueType::Int)
-        .filter_map(|attr| from_binary::<u64>(&attr.value).ok())
+        .flat_map(|page| &page.attributes)
+        .filter(|attr| attr.name == name && attr.attribute_type() == AttributeType::Int)
+        .filter_map(|attr| from_json::<u64>(&attr.value).ok())
         .map(Uint64::new)
         .collect()
 }
 
+/// Parses all group ids from the [Provenance Blockchain Attributes](https://docs.provenance.io/modules/account)
+/// provided by filtering for all values that match the given name and have an assigned int value.
+///
+/// # Parameters
+///
+/// * `attributes` Attributes fetched via a chain query.
+/// * `name` A [Provenance Blockchain Name Module](https://docs.provenance.io/modules/name-module)
+/// name used to write the attribute.
+pub fn get_group_id_attribute_values<S: Into<String>>(
+    attributes: &QueryAttributesResponse,
+    name: S,
+) -> Vec<Uint64> {
+    get_group_id_attribute_values_paginated(vec![attributes.clone()], name)
+}
+/// Generates a [name bind msg](MsgBindNameRequest) that will properly assign the given name value
+/// to a target address.  Assumes the parent name is unrestricted or that the contract has access to
+/// bind a name to the parent name.
+///
+/// # Parameters
+/// * `name` The dot-qualified name to use on-chain for name binding. Ex: myname.sc.pb will generate
+/// a msg that binds "myname" to the existing parent name "sc.pb".
+/// * `bind_to_address` The bech32 address to which the name will be bound.
+/// * `restricted` If true, the name will be bound as a restricted name, preventing future name
+/// bindings from using it as a parent name.
+pub fn msg_bind_name<S1: Into<String>, S2: Into<String>>(
```

Comment:
stolen from https://github.com/FigureTechnologies/funding-trading-bridge-smart-contract/blob/a92bafb4397360ac0a4febfbc8390c7a54080e84/src/util/provenance_utils.rs#L20 as it handles the parent/child name relationship nicely and replicates the old functionality that we lost from provwasm w/ the upgrade